gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinClusterUtils.java

```diff
@@ -63,8 +68,9 @@ public static Path getAppWorkDirPath(FileSystem fs, String applicationName, Stri
 
   public static Path getAppWorkDirPathFromConfig(Config config, FileSystem fs,
       String applicationName, String applicationId) {
-    if (config.hasPath(CLUSTER_WORK_DIR)) {
-      return new Path(config.getString(CLUSTER_WORK_DIR));
+    if (config.hasPath(GobblinClusterConfigurationKeys.CLUSTER_WORK_DIR)) {
+      return new Path(new Path(fs.getUri()), PathUtils.combinePaths(config.getString(GobblinClusterConfigurationKeys.CLUSTER_WORK_DIR),
```

Comment:
Shall we log the app-specific work directory here, since users may not expect applicationName / ID appearing ?

Reply:
Added logs on the caller side instead of inside the utility method.